### ISSUE TITLE
[optimal-quantifier-concatenation] false positive when using back reference

### ISSUE THREAD

**Issue by danielrentz** [ota-meshi/eslint-plugin-regexp#853]
**Information:**
- ESLint version: 9.31
- `eslint-plugin-regexp` version: 2.9.0

**Description**

Rule [optimal-quantifier-concatenation] triggers when using a capturing group with a back reference.

```ts
const rex = /(.).*\1/;
             ^^^^^ Error: '(.)' and '.*' can be combined into one quantifier '.+'.
```

**Comment by ota-meshi** [comment]
If you want to ignore it, change the option.
https://ota-meshi.github.io/eslint-plugin-regexp/rules/optimal-quantifier-concatenation.html#capturinggroups
